test/FunctionalTests/Microsoft.OData.Core.Tests/MessageWriterConcurrencyTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MessageWriterConcurrencyTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System;
+using Xunit;
+using Microsoft.Extensions.DependencyInjection;
+using System.Linq;
+
+namespace Microsoft.OData.Core.Tests
+{
+    public class MessageWriterConcurrencyTests
+    {
+        /// <summary>
+        /// Verifies that concurrent message writer does not interleave execution and isolates the underlying streams.
+        /// </summary>
+        /// <returns>A task for the asyncronous test</returns>
+
+        [Fact]
+        public async Task VerifyConcurrentResultsAreConsistentAsync()
+        {
+            ServiceCollection services = new();
+            services.AddDefaultODataServices();
+            ServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            await Task.CompletedTask;

```

Comment:
Why are we awaiting this completed task? Is it to ensure the code below is scheduled to run concurrently? Would `Task.Yield()` achieve the same goal, I see you're using it down below and it seems to express the intent more clearly.

Reply:
removed